.pre-commit-config.yaml

```diff
@@ -42,7 +42,7 @@ repos:
     rev: v0.8.3
     hooks:
     - id: ruff
-      args: ["--fix", "--show-fixes"]
+      args: ["--fix", "--show-fixes", "--extend-select=I"]
```

Comment:
Ideally, this is configured in a config file (such as `pyproject.toml`), rather than baked into the command in this way.

But, if I add `extend-select = ["I"]` to the top-level `pyproject.toml` file instead of this change, it doesn't work (no import sorting is performed). I'm not sure why that's the case.

On the other hand, `extend-select=["I"]` to the individual projects' `pyproject.toml`, such as `cuda_parallel/` or `cuda_experimental/` does work, but that wouldn't work to sort imports of the other `.py` files in the project. 

Any suggestions? @bdice perhaps you might know?

Reply:
I think there are some funny rules about how extend options are inherited? Try using a normal select instead of extend-select, or try using the “extend” option to control inheritance: https://github.com/rapidsai/cudf/blob/88d925187566cf187a9a4d5c4619e00e2a9fb9b1/python/cudf/pyproject.toml#L145

Also I think the “fix” (definitely) and “show fixes” (possibly) options can also be put into the config instead of passed as an arg here. I recommend keeping the pre-commit configuration argument-free so that running ruff on its own does the same thing as pre-commit. https://github.com/rapidsai/cudf/blob/88d925187566cf187a9a4d5c4619e00e2a9fb9b1/python/cudf_polars/pyproject.toml#L86

Reply:
Thanks! This fixed it:

>  try using the “extend” option to control inheritance

Reply:
By default ruff only uses a single pyproject.toml and it'll the "closest" one to the files that you are modifying and performs no extending of settings, so if you put settings into subproject pyproject.toml files it will completely override the settings from the parent directory's pyproject.toml. The extend option is how you tell it to merge them. ruff will ignore any pyproject.toml files that don't have a ruff section, which is why you still get proper linting with multiple pyproject.toml files but only a higher-level one containing a ruff config.

Reply:
OK - that aligns with my understanding. Thanks @vyasr and @bdice! Things are working as expected now.